libyul/AsmAnalysis.cpp

```diff
@@ -98,22 +98,22 @@ AsmAnalysisInfo AsmAnalyzer::analyzeStrictAssertCorrect(Dialect const& _dialect,
 
 vector<YulString> AsmAnalyzer::operator()(Literal const& _literal)
 {
-	expectValidType(_literal.type, _literal.location);
+	expectValidType(_literal.type, _literal.debugData->location);
 	if (_literal.kind == LiteralKind::String && _literal.value.str().size() > 32)
 		m_errorReporter.typeError(
 			3069_error,
-			_literal.location,
+			_literal.debugData->location,
 			"String literal too long (" + to_string(_literal.value.str().size()) + " > 32)"
 		);
 	else if (_literal.kind == LiteralKind::Number && bigint(_literal.value.str()) > u256(-1))
-		m_errorReporter.typeError(6708_error, _literal.location, "Number literal too large (> 256 bits)");
+		m_errorReporter.typeError(6708_error, _literal.debugData->location, "Number literal too large (> 256 bits)");
 	else if (_literal.kind == LiteralKind::Boolean)
 		yulAssert(_literal.value == "true"_yulstring || _literal.value == "false"_yulstring, "");
 
 	if (!m_dialect.validTypeForLiteral(_literal.kind, _literal.value, _literal.type))
 		m_errorReporter.typeError(
 			5170_error,
-			_literal.location,
+			_literal.debugData->location,
```

Comment:
Since we are refactoring, do we want to use `locationOf` in all these cases?

Reply:
I was thinking the same. Maybe we should do it that way.

Reply:
If we know the type, we should use the direct members.

Reply:
Ok, it might make sense now that we have one more member access in the chain.